packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -153,34 +155,87 @@ export class WaferMap extends FoundationElement {
     /**
      * @internal
      */
-    public render(): void {
-        this.renderQueued = false;
-        this.initializeInternalModules();
-        this.renderer?.drawWafer();
+    public setHoverData(
+        hoverWidth: number,
+        hoverHeight: number,
+        hoverOpacity: HoverDieOpacity,
+        hoverTransform: string
+    ): void {
+        this.hoverWidth = hoverWidth;
+        this.hoverHeight = hoverHeight;
+        this.hoverOpacity = hoverOpacity;
+        this.hoverTransform = hoverTransform;
     }
 
-    private queueRender(): void {
-        if (!this.$fastController.isConnected) {
-            return;
-        }
-        if (!this.renderQueued) {
-            this.renderQueued = true;
-            DOM.queueUpdate(() => this.render());
+    /**
+     * @internal
+     */
+    public update(): void {
+        if (this.waferMapUpdateTracker?.requiresContainerDimensionsUpdate) {
+            this.eventCoordinator?.detachEvents();
```

Comment:
Feels like repetitive code -  `this.eventCoordinator?.detachEvents()` / `this.eventCoordinator?.updateEvents()` . I think that these could be pulled out and called outside the `else if` chain.